### PR TITLE
Fix issues with usage of point2d, add support for various other peripherials

### PR DESCRIPTION
**Pull request by sanjay900**
Previously, Point2Ds were stored as pointers in a lot of collections, and this was causing issues as the pointer was staying the same and so things like the `pts` map were only ever storing one point. I changed this over to just storing Point2Ds as values, and then i was able to see all the points in the `pts` map.

Also, when constructing the outputs, i noticed that slightly different math was being used to construct outputs compared to the math used to construct the mapping to bhaptics. This caused issues with rounding, so i have changed the bhaptics logic to use the same math.

After that, i did some digging and figured out how to make it show up as a bunch of other accessories, such as  the x40, Tactal, TactGlove and Tactosy for arms / hands / feet.

I also added support for direct pwm instead of just the PCA9685, as for the other accessories you really don't want to need a PCA9685 board.

I also added the BLE2902 descriptor to the monitor characteristic, as this was necessary for using this on the quest.

I also changed the bluetooth mac address to use the one from the esp32 instead of hardcoding, as some of the accessories need to be built multiple times (such as a Tactosy for each arm) and hence a hardcoded mac address would not work.

I changed `src_filter` to `build_src_filter` as `src_filter` is now deprecated too.

I also found a nice JSON file while digging that lists every possible device, the ble appearance and the device name, so i added that in too as it is just useful to have.

With all these changes, i found that using the feedback tester built into bhaptics on android / windows / quest works.

### CODE REVIEW

**Comment by sanjay900** [comment]
image
image
image
imageimage
image

**Comment by leon0399** [comment]
Wow! I haven't though of this project for a while, and definitely not thought somebody will show interest for it! Thank you very much for your contribution! I'll check it as soon as I get home, presumably tomorrow!
Thank you one more time!

**Comment by sanjay900** [comment]
No problem mate, I've just done a aliexpress shop so i can actually build all of these lol, will be fun when that all arrives

**Review by leon0399 — CHANGES_REQUESTED** [review]

**Comment by leon0399** [comment]
I'm really surprised, how you managed to figure out how my code works  without any documentation and comments! Really great work there!

File: firmware/output_writers/pca9685.cpp [hunk 1]
```diff
@@ -2,5 +2,8 @@
 
 void PCA9685OutputWriter::writeOutput(outputIntensity_t intensity)
 {
+    if (intensity) {
+        printf("%d %d\n", this->num, intensity);
```
[the remark this answers is not in the document]
Comment (sanjay900) [reply firmware/output_writers/pca9685.cpp:6]: haha yep, ill do that now

**Comment by leon0399** [comment]
Perfect! If you have any further questions or just feeling to discuss anything feel free to contact me on Discord: leon0399#7352 or Telegram: @leon0399

**Comment by leon0399** [comment]
@sanjay900 how's your AliExpress shopping? Would love to see your approach to hardware! I will be really grateful if you can share your progress in discord and we can discuss which hardware work the best!

**Comment by sanjay900** [comment]
At the moment im waiting on some mates that are doing motor testing and they are waiting for motors to arrive, however my motor controllers and some little vibration motors have arrive, and so have all my esp32s. Im busy working on another project at the moment though so i haven't really been putting much into this currently

**Comment by leon0399** [comment]
> At the moment im waiting on some mates that are doing motor testing and they are waiting for motors to arrive

Sounds awesome! I'll be really happy, if you share results of your motor experiments! 🙏 Choosing right motor is really important for feedback quality! 🤞

**Pull request merged**